web3.js/src/connection.ts

```diff
@@ -1368,6 +1368,63 @@ export type StakeActivationData = {
   inactive: number;
 };
 
+/**
+ * Data slice argument for getProgramAccounts
+ */
+export type DataSlice = {
+  /** offset of data slice */
+  offset: number;
+  /** length of data slice */
+  length: number;
+};
+
+/**
+ * Memory comparison filter for getProgramAccounts
+ */
+export type MemcmpFilter = {
+  memcmp: {
+    /** offset into program account data to start comparison */
+    offset: number;
+    /** data to match, as base-58 encoded string and limited to less than 129 bytes */
+    bytes: string;
+  };
+};
+
+/**
+ * Data size comparison filter for getProgramAccounts
+ */
+export type DataSizeFilter = {
+  /** Size of data for program account data length comparison */
+  dataSize: number;
+};
+
+/**
+ * A filter object for getProgramAccounts
+ */
+export type GetProgramAccountsFilter = MemcmpFilter | DataSizeFilter;
+
+/**
+ * Configuration object for getProgramAccounts requests
+ */
+export type GetProgramAccountsConfig = {
+  /** Optional commitment level */
+  commitment?: Commitment;
+  /** Optional encoding for account data (default base64) */
+  encoding?: 'base64' | 'jsonParsed';
+  /** Optional data slice to limit the returned account data */
+  dataSlice?: DataSlice;
+  /** Optional array of filters to apply to accounts */
+  filters?: GetProgramAccountsFilter[];
+};
+
+/**
+ * Configuration object for getParsedProgramAccounts
+ */
+export type GetParsedProgramAccountsConfig = Exclude<
+  GetProgramAccountsConfig,
+  'encoding' | 'dataSlice'
```

Comment:
flow is puking on this `Exclude` keyword when trying to use web3.js since this commint in the spl-token js bindings.  Is there some magic to make it work?  cc/ @jstarry

Reply:
@t-nelson hmm, taking a look

Reply:
Thanks!  STR are just to bump web3 in spl-token/js and `npm run flow`.

Reply:
Fix here: https://github.com/solana-labs/solana/pull/17049